src/lite-yt-embed.js

```diff
@@ -28,7 +29,10 @@ class LiteYTEmbed extends HTMLElement {
          * TODO: Consider using webp if supported, falling back to jpg
          */
         if (!this.style.backgroundImage) {
-          this.style.backgroundImage = `url("https://i.ytimg.com/vi/${this.videoId}/hqdefault.jpg")`;
+            let browserSupportsWebP = LiteYTEmbed.support_format_webp();
+            let imageExtension = browserSupportsWebP ? "webp" : "jpg";
+            let imagePath = browserSupportsWebP ? "vi_webp" : "vi";
+            this.style.backgroundImage = `url("https://i.ytimg.com/${imagePath}/${this.videoId}/hqdefault.${imageExtension}")`;
```

Comment:
btw have you done much testing on the availability of the webp thumbnail for the various 'sizes' ?

relevant: https://github.com/paulirish/lite-youtube-embed/blob/master/youtube-thumbnail-urls.md

When I chose the default of `hqdefault` here, i had tested a few playlists' worth of videos and found that hqdefault was the only size that reliably had a defined thumbnail. but... i'd be REALLY happy to find out that there's always a `sddefault.webp` for everrry video.

Reply:
Also.. there was a less involved commit doing similar work here: https://github.com/paulirish/lite-youtube-embed/pull/130/commits/06657bd748a4b8a71528c1576bcd0a1f1686dbb6 Not sure I want to use it. Just crosslinking for my personal reference

Reply:
Thanks for the comments, you are right.

I will check the availability requirements shortly and improve the WebP detection logic.

Reply:
> I will check the availability requirements shortly and improve the WebP detection logic.

no need!  i ended up doing the same the other day.. :)

the "google i/o" paragraph of https://github.com/paulirish/lite-youtube-embed/blob/master/youtube-thumbnail-urls.md#aka-the-thumbnail-poster-image-etc is still correct. but, now with webp in the picture.. here's how i see the lay of the land:

* very old videos only have the hq_JPG
* theres a handful that have the hq_JPG _and_ the hq_WEBP.
* there's an equal handful that have the hq_JPG and the sd_JPG
* and most videos from the past couple years have all four. 

(note that SD is better than HQ for these things. also in this go-around i didnt look at maxresdefault. that seems like overkill for our usecase)

so.. perhaps the best logic (if done in JS) would be... **try the sd_webp first. if that fails, then just go back to the hq_jpg**.    Of course, this approach might be different with `<picture>` or `-webkit-image-set()` where we have built-in type selection.

Reply:
Hi Paul, sorry for late replying here. Please give me some more time to come up with a suggested plan. I am planning to have a chat with the Thunbnail Engineer team to get a good understanding of what thumbnail variants always exist or not.

Reply:
I've added a UA detection to decide when to serve WebP and used the following image thumbnails:

- In the case Background Image the selected option is `sddefault` and the Developer can customize the thumbnail variant if desired by using the `thumbnail` attribute.
- in the case of Picture Tag being detected inside the lite-youtube element, no background is being created, and pro users can use the Picture tag with FetchPriority to load the Poster image super early.